README.md

```diff
@@ -1 +1,242 @@
+
+```ruby
+include Pundit
+include Wor::Paginate
+extend SimpleCrud
+
+before_action :set_params
+
+def set_params
+  SimpleCrudController.params = params
+end
+
+rescue_from ActiveRecord::RecordNotFound, with: :not_found
+rescue_from ActiveRecord::StatementInvalid, with: :unprocessable
+rescue_from ActiveRecord::RecordInvalid, with: :unprocessable
+rescue_from ActionController::ParameterMissing, with: :unprocessable
+rescue_from Pundit::NotAuthorizedError, with: :forbidden
+
+def not_found(_exception)
+  head :not_found
+end
+
+def unprocessable(exception)
+  render json: { errors: exception }, status: :unprocessable_entity
+end
+
+def forbidden(exception)
+  render json: { errors: exception }, status: :forbidden
+end
+```
+
+### Each controller
+In case you need either update or create, create a method with the valid input params using standard rails:
+```ruby
+def author_params
+  params.permit(:first_name, :email, :last_name, :institution, :role)
+end
+```
+
+For the actual crud methods, just use the lines you need
+```ruby
+simple_crud_for :update
+simple_crud_for :show
+simple_crud_for :index
+simple_crud_for :create
+simple_crud_for :destroy
+```
+or rails-like syntax
+
+```ruby
+simple_crud_for only: [:update, :show]
+```
+
+or the following for all methods with default options:
+```ruby
+simple_crud
+```
+
+Each method supports different options, as in:
+```ruby
+simple_crud_for :index, paginate: false, authorize: false, serializer: CustomSerializer
+```
+
+- Paginate: whether it should paginate or not. `true` paginates using wor-paginate, `false` doesn't paginate at all.
+- Authorize: whether it should use Pundit to automatically check if the action is permitted. `true` uses pundit, `false` doesn't.
+- Authenticate: whether it should use Devise to check for a current_user. `true` authenticates, `false` doesn't.
+- Serializer: specify a particular serializer you should use. If none is supplied, the default one for the model will be used.
+- Filter: specify whether index should use toscha-filterable to filter.  `true` filters, `false` doesn't.
+
+You'll need a few things so they work correctly:
+
+### Options
+#### Paginate
+No parameters are needed, but the pagination is done using [wor-paginate](https://github.com/Wolox/wor-paginate) so read the documentation in case you need to customize the output.
+
+#### Authorize
+The name of the policy should be the model followed by Policy, as in `AuthorPolicy`. Support for custom policies is upcoming. The policy should be a standard Pundit policy, for example:
+
+```ruby
+class AuthorPolicy
+  attr_reader :user, :author
+
+  def initialize(user, author)
+    @user = user
+    @author = author
+  end
+
+  def show?
+    user.present?
+  end
+end
+
+```
+
+#### Authenticate
+SimpleCrud will assume a current_user method. Future versions will support a custom model. Defining a current_user method in ApplicationController should work if you're using a different model, as of now.
```

Comment:
you could have a `simple_crud_user` method, which defaults to current_user, just like pundit does, if you define a `pundit_user`, it will be used by pundit.

Reply:
BTW, there is no indication the `authorize: true` is required to use policies, if it's not, I miss something at the code I can't catch.